addons/account/migrations/13.0.1.1/post-migration.py

```diff
@@ -310,50 +325,31 @@ def migration_invoice_moves(env):
                 OR ail.account_analytic_id = aml.analytic_account_id)
             AND aml.tax_line_id IS NULL
             AND aml.old_invoice_line_id IS NULL
-        RETURNING aml.id""",
+            """,
+    )
+    # Adding all the missing lines
+    openupgrade.logged_query(
+        env.cr, """
+        INSERT INTO account_move_line (company_id, journal_id, account_id,
+        exclude_from_invoice_tab, sequence, name, quantity, price_unit, discount,
+        price_subtotal, price_total, company_currency_id, currency_id, partner_id, product_uom_id,
+        product_id, analytic_account_id, display_type, is_rounding_line,
+        move_id, old_invoice_line_id, date, create_uid, create_date, write_uid,
+        write_date, parent_state, move_name, credit, debit, balance)
+        SELECT ail.company_id, am.journal_id, ail.account_id, FALSE, ail.sequence, ail.name,
+        ail.quantity, ail.price_unit, ail.discount, ail.price_subtotal,
+        ail.price_total, rc.currency_id, CASE WHEN rc.currency_id != ail.currency_id
+        THEN ail.currency_id ELSE NULL END, ail.partner_id, ail.uom_id,
+        ail.product_id, ail.account_analytic_id, ail.display_type,
+        ail.is_rounding_line, COALESCE(ai.move_id, am.id), ail.id, COALESCE(ai.date, ai.date_invoice),
+        ail.create_uid, ail.create_date, ail.write_uid, ail.write_date, am.state, am.name,
+        0.0, 0.0, 0.0
+        FROM account_invoice_line ail
+            JOIN account_invoice ai ON ail.invoice_id = ai.id
+            JOIN account_move am ON am.old_invoice_id = ai.id
+            LEFT JOIN res_company rc ON ail.company_id = rc.id
+        WHERE ail.aml_matched is not TRUE""",
     )
-    aml_ids2 = tuple(x[0] for x in env.cr.fetchall())
-    if aml_ids2:
-        openupgrade.logged_query(
-            env.cr, """
-            SELECT ail.id
-            FROM account_invoice_line ail
-                JOIN account_invoice ai ON ail.invoice_id = ai.id AND ai.state NOT IN ('draft', 'cancel')
-                JOIN account_move am ON ail.invoice_id = am.old_invoice_id
-                JOIN account_move_line aml ON am.id = aml.move_id
-            WHERE ail.company_id = aml.company_id AND ail.account_id = aml.account_id
-                AND ail.partner_id = aml.partner_id
-                AND ((ail.product_id IS NULL AND aml.product_id IS NULL) OR ail.product_id = aml.product_id)
-                AND ((ail.uom_id IS NULL AND aml.product_uom_id IS NULL) OR ail.uom_id = aml.product_uom_id)
-                AND ((ail.account_analytic_id IS NULL AND aml.analytic_account_id IS NULL)
```

Comment:
That might give you problems if your system edited the following function
https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_invoice.py#L1234

It might be better to import all the data if possible

Reply:
Do you think it could be possible to define at the beginning something like : 
hash_keys = ["account_id", "tax_ids", "tax_line_id", "product_id", "analytic_account_id", "date_maturity", "analytic_tag_ids"]

and then generate an sql query based on the values ? 
so user that has overloaded the function could more easily write the correct settings.

(not sure if what I say is relevant : never migrated from 12.0 to 13.0 for the time being).

Reply:
I think it is not relevant. The point is to keep all the data as it was on the invoice. In my casa, I removed a key on that hook and migration removed all the lines except one